frontend/src/components/project/task.jsx

```diff
@@ -1,19 +1,74 @@
-import React from 'react';
+import React, {PropTypes} from 'react';
+import {ProjectActions} from '../../actions/actions';
+import {ItemTypes} from '../../constants/dragAndDropConstants';
+import {DragSource, DropTarget} from 'react-dnd';
+
+const taskSource = {
+  beginDrag(props) {
+    return {id: props.id};
+  }
+};
+
+const taskTarget = {
+  // this event will happen frequently, on any movement while hovering
+  // over another task
+  hover(props, monitor) {
+    const draggedId = monitor.getItem().id;
+    if (draggedId !== props.id) {
+      ProjectActions.updateTaskRankLocally({
+        draggedTaskId: draggedId,
+        targetTaskId: props.id
+      });
+    }
+  },
+
+  // on dropping a task card on another task card, a message should be sent to the server
```

Comment:
I am an idiot! I need to check whether I am dropping a task on itself (as will happen after tasks have swapped during hover) here, and fire an action if the task is dropped on another task and not on itself. Instead, I was doing unnecessary logic in the ProjectStore. I'll fix that.